### ISSUE TITLE
Create Resource related data when accessed

### ISSUE THREAD

**Issue by developerium** [marmelab/react-admin#3136]
Hi guys

I have an api with over 1000 resources which I'm registering `Resource`s dynamically, this has caused a performance issue for me.
The user obviously won't be able to use all of them at once, maybe at most 50 of them at a time, so all other resources will occupy space which will slow redux changes down.
Is there any way we can make resource related data in redux on fly, whenever access by user?

**Comment by Kmaschta** [comment]
Ouch. We've never tried to manage a that big number of resources.

The react admin API rely on a declarative API, it's very unlikely that core maintainers of the lib will implement a dynamic resource declaration because we have not that need, and this is not a common need neither.

In any case, you will have to implement your own custom system, but just for the sake of the experiment I'm curious to see the code as well!

That said, I'd recommend [Stack Overflow](https://stackoverflow.com/questions/tagged/react-admin) to seek help about this kind of specific problem.

**Comment by fzaninotto** [comment]
You can dynamically set resources based on authorization, see the related doc:

https://marmelab.com/react-admin/Authorization.html#restricting-access-to-resources-or-views

Other use cases with such a large number of resources are not supported.

**Comment by developerium** [comment]
Ok thanks, I will look into this when I have some spare time and see if I can add the functionality

**Comment by developerium** [comment]
For anyone interested, I managed to work around this and thought share it here.

These are the steps I took:
- I only registered one essential route with Resource component.
- when user loges in, user sees the dashboard and menu
- I registered **customRoutes** and I got resource name the custom page component that will handle that pattern.
- custom page will check if this resource name is not registered, will dispatch an action to keep it in a list in redux store
- we also need a component on layout (because we always need it), and this component will check for every resource in the mentioned list, and will render **Resource** based on them
- this will trigger Resource component making the necessary store properties 

This steps may seem too much, but when you implement it, it's not that hard.

Cheers

**Comment by WiXSL** [comment]
@developerium, I'm trying to follow your implementation but i don't understand what you are saying.

- "`I only registered one essential route with Resource component.`"
**You create only one resource component in your `Admin` right?**

- "`I registered customRoutes and I got resource name the custom page component that will handle that pattern.`"
**I couldn't understand what you are saying here. You register in customRoutes a route for each of your views?
How you relate the resource with this customRoutes?**
